var/spack/repos/builtin/packages/charliecloud/package.py

```diff
@@ -10,8 +10,10 @@ class Charliecloud(MakefilePackage):
     """Lightweight user-defined software stacks for HPC."""
 
     homepage = "https://hpc.github.io/charliecloud"
-    url      = "https://github.com/hpc/charliecloud/archive/v0.2.4.tar.gz"
+    url = "https://github.com/hpc/charliecloud/archive/v0.2.4.tar.gz"
+    git = "https://github.com/hpc/charliecloud.git"
 
+    version('0.9.10', preferred=True, sha256='44e821b62f9c447749d3ed0d2b2e44d374153058814704a5543e83f42db2a45a')
```

Comment:
You don't need `preferred` here as Spack already defaults to the latest version.